src/app_service/service/keycardV2/dto.nim

```diff
@@ -0,0 +1,169 @@
+import json, strutils
+include ../../common/json_utils
+
+
+
+type StateString* = enum
+  UnknownReaderState = "unknown"
+  NoPCSC = "no-pcsc"
+  InternalError = "internal-error"
+  WaitingForReader = "waiting-for-reader"
+  WaitingForCard = "waiting-for-card"
+  ConnectingCard = "connecting-card"
+  ConnectionError = "connection-error" # NOTE: Perhaps a good place for retry
+  NotKeycard = "not-keycard"
+  PairingError = "pairing-error"
+  EmptyKeycard = "empty-keycard"
+  NoAvailablePairingSlots = "no-available-pairing-slots"
+  BlockedPIN = "blocked-pin" # PIN remaining attempts == 0
+  BlockedPUK = "blocked-puk" # PUK remaining attempts == 0
+  FactoryResetting = "factory-resetting"
+  Ready = "ready"
+  Authorized = "authorized"
+
+type KeycardState* = enum
+  UnknownReaderState = -1,
+  NoPCSCService,
+  PluginReader,
+  InsertKeycard,
+  ReadingKeycard,
+  WrongKeycard,
+  NotKeycard,
+  MaxPairingSlotsReached,
+  Locked,
+  NotEmpty,
+  Empty,
+  Authorized
+
+type KeycardInfoDto* = object
+  initialized*: bool
+  instanceUID*: string
+  version*: string
+  availableSlots*: int
+  keyUID*: string
+
+type KeycardStatusDto* = object
+  remainingAttemptsPIN*: int
+  remainingAttemptsPUK*: int
+  keyInitialized*: bool
+  path*: string
+
+type KeycardEventDto* = object
+  state*: KeycardState
+  keycardInfo*: KeycardInfoDto
+  keycardStatus*: KeycardStatusDto
+
+type
+  KeyDetailsV2* = object
+    address*: string
+    publicKey*: string
+    privateKey*: string
+
+type KeycardExportedKeysDto* = object
+  eip1581Key*: KeyDetailsV2
+  encryptionKey*: KeyDetailsV2
+  masterKey*: KeyDetailsV2
+  walletKey*: KeyDetailsV2
+  walletRootKey*: KeyDetailsV2
+  whisperKey*: KeyDetailsV2
+  masterKeyAddress*: string
+
+proc fromStringStateToInt*(state: StateString): KeycardState =
+  case state
+  of StateString.UnknownReaderState:
+    result = KeycardState.UnknownReaderState
+  of StateString.NoPCSC:
+    result = KeycardState.NoPCSCService
+  of StateString.InternalError:
+    result = KeycardState.UnknownReaderState
+  of StateString.WaitingForReader:
+    result = KeycardState.PluginReader
+  of StateString.WaitingForCard:
+    result = KeycardState.InsertKeycard
+  of StateString.ConnectingCard:
+    result = KeycardState.ReadingKeycard
+  of StateString.ConnectionError:
+    result = KeycardState.WrongKeycard # TODO figure out if this is correct
+  of StateString.NotKeycard:
+    result = KeycardState.NotKeycard
+  of StateString.PairingError:
+    result = KeycardState.NotEmpty # TODO figure out if this is correct
```

Comment:
As I described in https://github.com/status-im/status-desktop/issues/17140, `PairingError` means that the keycard has non-default pairing password. We should add a view for it.